indico_toolkit/indico_wrapper/modelop.py

```diff
@@ -0,0 +1,204 @@
+from indico import IndicoClient
+from indico.queries import GraphQLRequest
+import json
+import string
+
+
+class ModelOp:
+
+    """
+    Class to assit in Model Group Operation calls.
+    """
+
+    def __init__(self, client: IndicoClient):
+        self.client = client
+        self.task_type = None
+
+    def get_model_group(self, model_group_id: int) -> dict: 
+        """
+        Get specified model group information.
+        Args:
+            model_group_id (int): id of model group 
+        Returns:
+            dict: attributes of the specified model group id 
+        """
+
+        query = """
+        
+query GetModelGroup($id: Int) {
+  modelGroups(modelGroupIds: [$id]) {
+    modelGroups {
+      id
+      name
+      status
+      taskType
+      selectedModel {
+        id
+        status
+      }
+      modelOptions {
+        id
+        samplingStrategy
+        modelTrainingOptions
+      }
+    }
+  }
+}
+                """
+        
+        results = self.client.call(
+            GraphQLRequest(query, {"modelGroupId": model_group_id})
+        )
+        models = results["modelGroups"]["modelGroups"]
+        for model in models:
+            if model["id"] == model_group_id:
```

Comment:
I imagine the goal here would be to get the settings of the most recent model ID or a specific model ID-- i.e. you'd probably want to be matching against a Model ID (vs. the model group ID which all of these models should be a part of by default)

Reply:
this query pulls all model groups hence this filtering. maybe the query can be altered to only retrieve the specific model group called. 

the actual model id queried from this call is the last selected model id, which doesn't account for the new model id that is generated after retraining. this method could be run after retraining and should show the most recent model (assuming the model was successful in retraining).

Reply:
It pulls all model groups? Or it pulls all model IDs that are part of a specific model group?

Reply:
I'd be surprised if it's returning all model groups in the cluster-- but I think my point holds

Reply:
Pulls all model groups. Seems redundant since `list_models` pulls all the model IDs for a specified model group.

Reply:
Combined the two functions. With a specified Model Group ID, it returns all models. If a specific model id is also passed, it only returns that model.